api/src/pcapi/core/offers/api.py

```diff
@@ -804,22 +828,22 @@ def edit_stock(
                 id_at_provider,  # type: ignore[arg-type]
                 stock.id,
             )
-        modifications["idAtProviders"] = id_at_provider
+        modifications["idAtProviders"] = FieldUpdateLog(old=stock.idAtProviders, new=id_at_provider)
 
     if not modifications:
         logger.info(
             "Empty update of stock",
             extra={"offer_id": stock.offerId, "stock_id": stock.id},
         )
-        return None, False  # False is for `"beginningDatetime" in modifications`
+        return None, None
 
     if stock.offer.isFromAllocine:
         updated_fields = set(modifications)
         validation.check_update_only_allowed_stock_fields_for_allocine_offer(updated_fields)
         stock.fieldsUpdated = list(set(stock.fieldsUpdated) | updated_fields)
 
     for model_attr, value in modifications.items():
-        setattr(stock, model_attr, value)
+        setattr(stock, model_attr, value.new)
```

Comment:
je pense que le mieux ça serait de garder le fonctionnement actuelle, donc que que `modifications` reste un dict avec juste la nouvelle valeur et ici tu fait quelque chose comme :
```suggestion
        # changes un dict à initialiser  avant
        changes[model_attr] = ColumnUpdateLog(old_value=getattr(stock, model_attr), new_value=value)
        setattr(stock, model_attr, value)
```